src/components/Activity.tsx

```diff
@@ -1,15 +1,24 @@
 import ActivityCard from './ActivityCard'
 import { useAllActivities } from '../hooks/useAllActivities';
+import { CircularProgress } from '@mui/material';
 
 
 const Activity: React.FC = () => {
-  const allActivities = useAllActivities().data;
+  const allActivities = useAllActivities().apiResponse.data;
+  const loading = useAllActivities().loading;
```

Comment:
usually when we are checking on a boolean we name it with a verb, like `isLoading`

Reply:
It can be written like this also:  
  const { apiResponse: allActivities } = useAllActivities();
 const { isLoading } = useAllActivities();

and can be used below as
allActivities.data

Reply:
Changed 'loading' to 'isLoading'
and rewrote the apiResponse like you suggested

Reply:
actually they can be both in one line, like: 
const { apiResponse: allActivities, isLoading } = useAllActivities();

just saw it :)

Reply:
Nice, changed it to one line :)